agent/reconciler/host_reconciler.go

```diff
@@ -94,12 +98,28 @@ func (r *HostReconciler) Reconcile(ctx context.Context, req ctrl.Request) (_ ctr
 
 func (r *HostReconciler) reconcileNormal(ctx context.Context, byoHost *infrastructurev1beta1.ByoHost) (ctrl.Result, error) {
 	logger := ctrl.LoggerFrom(ctx)
+	logger = logger.WithValues("ByoHost", byoHost.Name)
+	logger.Info("reconcile normal")
 	if byoHost.Status.MachineRef == nil {
 		logger.Info("Machine ref not yet set")
 		conditions.MarkFalse(byoHost, infrastructurev1beta1.K8sNodeBootstrapSucceeded, infrastructurev1beta1.WaitingForMachineRefReason, clusterv1.ConditionSeverityInfo, "")
 		return ctrl.Result{}, nil
 	}
 
+	if r.UseInstallerController && !conditions.IsTrue(byoHost, infrastructurev1beta1.K8sComponentsInstallationSucceeded) {
+		if byoHost.Spec.InstallationSecret == nil {
+			logger.Info("InstallationSecret not ready")
+			conditions.MarkFalse(byoHost, infrastructurev1beta1.K8sComponentsInstallationSucceeded, infrastructurev1beta1.K8sInstallationSecretUnavailableReason, clusterv1.ConditionSeverityInfo, "")
+			return ctrl.Result{}, nil
+		}
+		err := r.executeInstallerController(ctx, byoHost)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+		r.Recorder.Event(byoHost, corev1.EventTypeNormal, "InstallScriptExecutionSucceeded", "install script executed")
+		conditions.MarkTrue(byoHost, infrastructurev1beta1.K8sComponentsInstallationSucceeded)
+	}
+
```

Comment:
let's move this logic to [:119](https://github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/blob/main/agent/reconciler/host_reconciler.go#L119) as @anusha94 has mentioned [here](https://github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/pull/564/files#r883285864)

Reply:
I have moved inside if..else block. Thanks